library/src/hcc_detail/hipsolver.cpp

```diff
@@ -209,6 +211,27 @@ hipsolverStatus_t rocblas2hip_status(rocblas_status_ error)
     }
 }
 
+#define CHECK_ROCBLAS_ERROR(STATUS)             \
+    do                                          \
+    {                                           \
+        rocblas_status _status = (STATUS);      \
+        if(_status != rocblas_status_success)   \
+            return rocblas2hip_status(_status); \
+    } while(0)
+
+inline rocblas_status hipsolverManageWorkspace(rocblas_handle handle, int lwork)
+{
+    size_t current_size;
+    if(rocblas_is_user_managing_device_memory(handle))
+        rocblas_get_device_memory_size(handle, &current_size);
+
+    if(lwork > current_size)
+        return rocblas_set_device_memory_size(handle, lwork);
```

Comment:
`lwork` is user-supplied, and is implicitly converted from `int` to `size_t` for this call. If the user has supplied a negative value, the result will be an allocation with a nonsense size (probably very large). That might be fine. It's garbage-in-garbage-out, after all. Just checking whether that's the intended design of the API.

Even if there's no validity check being done, I would recommend making the cast to `size_t` explicit. It will make the conversion more obvious to future developers, and you can avoid a signed / unsigned comparison by putting it before the `lwork > current_size` comparison and using the converted value. The C++ integer promotion rules are surprisingly complicated, so it's nice to avoid the complexity by matching types exactly and making conversions explicit.